src/trunk.c

```diff
@@ -8964,39 +8965,52 @@ trunk_config_init(trunk_config        *trunk_cfg,
    // Inline what we would get from trunk_pivot_size(trunk_handle *).
    trunk_pivot_size = data_cfg->max_key_size + sizeof(trunk_pivot_data);
 
-   // Setting hard limit and over overprovisioning
+   // Setting hard limit and check configuration for over-provisioning
```

Comment:
I don't understand this change. It looks like we're setting the overprovision, not checking for it.

Reply:
I interpreted the comment as follows:

`Setting hard limit` applies to stuff like this we do further below in this code blow:

```
9015    // Internally determined hard-limit, which effectively depends on the
9016    // - configured page size and trunk header size
9017    // - user-specified configured key size
9018    // - user-specified fanout
9019    trunk_cfg->hard_max_branches_per_node =
9020       bytes_for_branches / sizeof(trunk_branch) - 1;
```

The `check configuration for over-provisioning` is referring to these config params which are not practically supportable for the given page-size. This is exactly what this change is fixing, i.e., error checks of this type:

```
8982    // Deal with mis-configurations where we don't have available bytes per
8983    // pivot key
8984    uint64 available_bytes_per_pivot_key = 0;
8985    if (available_bytes_per_pivot > sizeof(trunk_pivot_data)) {
8986       available_bytes_per_pivot_key =
8987          available_bytes_per_pivot - sizeof(trunk_pivot_data);
8988    }
8989
8990    if (trunk_node_min_size >= page_size) {
8991       platform_error_log("Trunk node min size=%lu bytes "
```

That's why I thought `check configuration for over-provisioning' was reasonably accurate.

I'm marking this as resolved; Do suggest a reword if you still think it's incorrect or could be improved.